lib/network.go

```diff
@@ -1914,6 +1915,15 @@ type MsgDeSoHeader struct {
 	// event that ASICs become powerful enough to have birthday problems in the future.
 	ExtraNonce uint64
 
+	// TransactionsConnectStatus is only used for Proof of Stake blocks, starting with
+	// MsgDeSoHeader version 2. For all earlier versions, this field will default to nil.
+	//
+	// This bitset field stores information whether each transaction in the block passes
+	// or fails to connect. The bit at i-th position is set to 1 if the i-th transaction
+	// in the block passes connect, and 0 otherwise. The length of the bitset must match
+	// the number of transactions in the block.
+	TransactionsConnectStatus *bitset.Bitset
```

Comment:
Please move the `TransactionsConnectStatus` field out of MsgDeSoHeader and into MsgDeSoBlock. The txn statuses are most relevant to the txns in the block at the time the txns are being connected. 

You can parse the field in MsgDeSoBlock.FromBytes() based on the block header's version. A good example is how the MsgDeSoBlock.BlockProducerInfo field is parsed based on the block header version. The block header's version is used as a proxy for the entire block's version. 

You can follow the same pattern, and check for header version >= 2 here: https://github.com/deso-protocol/core/blob/feature/proof-of-stake/lib/network.go#L2637